src/main/scala/EnsimeImplicitPlugin.scala

```diff
@@ -41,20 +41,35 @@ class EnsimeImplicitPlugin(override val global: Global) extends Plugin {
     override val phaseName: String = "null-cachedImplicit"
 
     // best way to inspect a tree, just call this
-    def debug(name: String, tree: Tree): Unit = {
-      global.reporter.warning(tree.pos, s"$name: ${showCode(tree)}\n${showRaw(tree)}")
+    private def debug(name: String, tree: Tree): Unit = {
+      println(s"$name ${tree.id} ${tree.pos}: ${showCode(tree)}\n${showRaw(tree)}")
     }
 
+    private def missingExplicitType(t: Tree) = reporter.error(t.pos, "Missing explicit type on implicit value when calling `cachedImplicit`")
+    private def notDoingAnyWork(t: Tree) = reporter.warning(t.pos, s"`cachedImplicit` is not doing any work (ensime plugin)")
+
     override def transform = {
-      case tree @ ValDef(mods, name, tpe, TypeApply(Ident(CachedImplicit), _)) =>
-        if (tpe == EmptyTree) {
-          reporter.error(tree.pos, s"Missing explicit type in call to `cachedImplicit`.")
+      case tree @ ValDef(mods, name, tpt: TypeTree, TypeApply(Ident(CachedImplicit), _)) =>
+        if (tpt.original == EmptyTree || (tpt.original eq null)) {
```

Comment:
I think this check isn't necessary, `cachedImplicit` has a `TypeApply` meaning there's an explicit type argument on the right-hand side.